clickhouse-jdbc/src/main/java/com/clickhouse/jdbc/PreparedStatementImpl.java

```diff
@@ -1,306 +1,397 @@
 package com.clickhouse.jdbc;
 
+import com.clickhouse.logging.Logger;
+import com.clickhouse.logging.LoggerFactory;
+
 import java.io.InputStream;
 import java.io.Reader;
 import java.math.BigDecimal;
 import java.net.URL;
+import java.nio.charset.StandardCharsets;
 import java.sql.*;
 import java.util.Calendar;
 
 public class PreparedStatementImpl extends StatementImpl implements PreparedStatement, JdbcWrapper {
-    String sql;
+    private static final Logger log = LoggerFactory.getLogger(PreparedStatementImpl.class);
+
+    String initialSql;
+    String [] sqlSegments;
+    Object [] parameters;
     public PreparedStatementImpl(ConnectionImpl connection, String sql) {
         super(connection);
-        this.sql = sql;
+        this.initialSql = sql;
+        //Split the sql string into an array of strings around question mark tokens
+        this.sqlSegments = sql.split("\\?");
```

Comment:
how would it handle `IN(?, ?, ?)` ? 
I think this logic is part of execution phase when we know more about parameters.